### PR TITLE
syncer: fix stuffs

### PR DESCRIPTION
**Pull request by missionliao**
- fix a bug that syncer won't work when **numChains** changed 8be8c04 5e51bb1
- fix a bug that a block is confirmed twice when sent to `core.NewConsensusFromSyncer`
- skip finalized block when delivered to `core.compactionChain` b174a14
- check block hash sequence delivered from the constructed `core.Lattice` instance when syncing.

### CODE REVIEW

**Comment by missionliao** [comment]
@jimmyhu-dexon  @haoping-ku  rebase `master` and fix conflicts.

File: core/syncer/consensus.go [hunk 1]
```diff
@@ -333,6 +355,25 @@ func (con *Consensus) findLatticeSyncBlock(
...
+		return nil
+	}
+	con.finalizedBlockHashes = append(con.finalizedBlockHashes, block.Hash)
+	delivered, err := con.lattice.ProcessFinalizedBlock(block)
+	if err != nil {
+		return err
+	}
+	for idx, b := range delivered {
+		if con.finalizedBlockHashes[idx] != b.Hash {
+			return test.ErrMismatchBlockHashSequence
```
Comment (jimmyhu-dexon) [inline core/syncer/consensus.go]: Don't return a value of test package in non-test package.

Comment (missionliao) [reply core/syncer/consensus.go]: Fixed

**Comment by missionliao** [comment]
Just found the unittest always failed after adjust the round-interval from 30s to 50s, investigating.

**Comment by missionliao** [comment]
@jimmyhu-dexon  @haoping-ku  Changes:
- add more logs.
- fix a bug that `restartNotary` is trigger by older blocks from pulling (we would go into `Lattice not ready` hell when this happened)
- avoid rebroadcasting block-randomness from syncer.

**Pull request merged**